### PR TITLE
Feat/fix staticcheck

### PR DESCRIPTION
**Pull request by coryschwartz**
Context: https://github.com/protocol/.github/issues/41

### CODE REVIEW

File: dial_test.go [hunk 1]
```diff
@@ -437,12 +437,13 @@ func TestDialBackoffClears(t *testing.T) {
...
 	before := time.Now()
-	if c, err := s1.DialPeer(ctx, s2.LocalPeer()); err == nil {
+	c, err := s1.DialPeer(ctx, s2.LocalPeer())
+	if err == nil {
 		t.Fatal("dialing to broken addr worked...", err)
-		defer c.Close()
 	} else {
 		t.Log("correctly got error:", err)
 	}
+	defer c.Close()
```
Comment (marten-seemann) [inline dial_test.go:446]: If `err != nil`, then `c` might be `nil`. The `defer` should be moved before the `t.Fatal`.

File: swarm_dial.go [hunk 2]
```diff
@@ -237,18 +237,18 @@ func (s *Swarm) DialPeer(ctx context.Context, p peer.ID) (network.Conn, error) {
...
-	var logdial = lgbl.Dial("swarm", s.LocalPeer(), p, nil, nil)
+	lgbl.Dial("swarm", s.LocalPeer(), p, nil, nil)
 	err := p.Validate()
 	if err != nil {
 		return nil, err
 	}
 
 	if p == s.local {
-		log.Event(ctx, "swarmDialSelf", logdial)
+		// log.Event(ctx, "swarmDialSelf", logdial)
```
Comment (marten-seemann) [inline swarm_dial.go:247]: @Stebalien, how are we supposed to deal with this?

File: swarm_dial.go [hunk 3]
```diff
@@ -237,18 +237,18 @@ func (s *Swarm) DialPeer(ctx context.Context, p peer.ID) (network.Conn, error) {
 // dialbackoff.
 func (s *Swarm) dialPeer(ctx context.Context, p peer.ID) (*Conn, error) {
 	log.Debugf("[%s] swarm dialing peer [%s]", s.local, p)
-	var logdial = lgbl.Dial("swarm", s.LocalPeer(), p, nil, nil)
+	lgbl.Dial("swarm", s.LocalPeer(), p, nil, nil)
```
Comment (Stebalien) [inline swarm_dial.go:240]: This doesn't actually do anything. We should replace these two lines with:

```go
log.Debugw("dialing peer", "from", s.local, "to", p)
```

File: swarm_dial.go [hunk 4]
```diff
@@ -237,18 +237,18 @@ func (s *Swarm) DialPeer(ctx context.Context, p peer.ID) (network.Conn, error) {
...
 	}
 
 	if p == s.local {
-		log.Event(ctx, "swarmDialSelf", logdial)
+		// log.Event(ctx, "swarmDialSelf", logdial)
 		return nil, ErrDialToSelf
 	}
 
-	defer log.EventBegin(ctx, "swarmDialAttemptSync", p).Done()
+	// defer log.EventBegin(ctx, "swarmDialAttemptSync", p).Done()
```
Comment (Stebalien) [inline swarm_dial.go:251]: Let's just delete these lines.

File: swarm_listen.go [hunk 5]
```diff
@@ -111,7 +111,7 @@ func (s *Swarm) AddListenAddr(a ma.Multiaddr) error {
 					// ignore.
 					return
 				default:
-					log.Warningf("add conn %s failed: ", err)
+					log.Warnf("add conn %s failed: ", err)
```
Comment (Stebalien) [inline swarm_listen.go:114]: If we're changing these, we might as well switch to a structured version:

```go
log.Warnw("adding connection failed", "to", p, "error", err)
```

**Review by Stebalien — CHANGES_REQUESTED** [review]

File: dial_test.go (hunk 1 above)
Comment (marten-seemann) [reply dial_test.go:446]: Fixed in #255.

File: swarm_dial.go (hunk 3 above)
Comment (marten-seemann) [reply swarm_dial.go:240]: Fixed in #255.

File: swarm_listen.go (hunk 5 above)
Comment (marten-seemann) [reply swarm_listen.go:114]: Fixed in #255.

File: swarm_dial.go (hunk 2 above)
Comment (Stebalien) [reply swarm_dial.go:247]: `log.Debugw` with a message.

**Pull request closed without merging**